plugins/templates/create.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const hoek = require('hoek');
+const urlLib = require('url');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/templates',
```

Comment:
Template is not any different from `pipeline` or other models. For example, `scmUri` is a unique field of a pipeline. There cannot be any two pipelines with the same `scmUri`. Same as `job`. There cannot be any two jobs with the same `pipelineId +  name`

I'm not opposing to having an endpoint to get by name and version (we need it). But I don't think there is any harm in having a `GET /{id}` since we do have this endpoint consistently for ALL the models.